compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -2894,9 +2842,12 @@ where
                 let max_by_val_size = Pointer.size(cx) * 2;
                 let size = arg.layout.size;
 
-                if arg.layout.is_unsized() || size > max_by_val_size {
-                    arg.make_indirect();
-                } else {
+                assert!(
+                    matches!(arg.mode, PassMode::Indirect { on_stack: false, .. }),
```

Comment:
```suggestion
                    let something = matches!(arg.mode, PassMode::Indirect { on_stack: false, .. });
```
Nit: this would be better for formatting and readability is the match result was given a name.